experiments/yarn/src/main/scala/org/apache/gearpump/experiments/yarn/ResouceManagerClientActor.scala

```diff
@@ -1,44 +1,47 @@
 package org.apache.gearpump.experiments.yarn
 
-import org.apache.gearpump.experiments.yarn.Actions.AMStatusMessage
-import org.apache.gearpump.experiments.yarn.Actions.ContainerRequestMessage
-import org.apache.gearpump.experiments.yarn.Actions.RegisterAMMessage
-import org.apache.gearpump.experiments.yarn.master.ResourceManagerCallbackHandler
-import org.apache.gearpump.experiments.yarn.master.YarnApplicationMaster.TIME_INTERVAL
+import akka.actor.{Actor, actorRef2Scala}
+import org.apache.gearpump.experiments.yarn.master.{AmActorProtocol, ResourceManagerCallbackHandler, YarnApplicationMaster}
 import org.apache.gearpump.util.LogUtil
-import org.apache.hadoop.yarn.api.records.Priority
-import org.apache.hadoop.yarn.api.records.Resource
+import org.apache.hadoop.yarn.api.records.{Priority, Resource}
 import org.apache.hadoop.yarn.client.api.AMRMClient.ContainerRequest
 import org.apache.hadoop.yarn.client.api.async.AMRMClientAsync
 import org.apache.hadoop.yarn.conf.YarnConfiguration
 import org.apache.hadoop.yarn.util.Records
-import akka.actor.Actor
-import akka.actor.ActorRef
-import akka.actor.actorRef2Scala
-import org.apache.gearpump.experiments.yarn.master.YarnApplicationMaster
 
-class ResourceManagerClientActor(yarnConf: YarnConfiguration, yarnAM: ActorRef) extends Actor {
+import scala.util.{Failure, Success, Try}
+
+class ResourceManagerClientActor(yarnConf: YarnConfiguration, clientFactory : AMRMClientAsyncFactory[ContainerRequest]) extends Actor {
+
+  import AmActorProtocol._
+
   val LOG = LogUtil.getLogger(getClass)
   var client: AMRMClientAsync[ContainerRequest] = _
   
   override def receive: Receive = {
     case rmCallbackHandler: ResourceManagerCallbackHandler =>
       LOG.info("Received RMCallbackHandler")
-      client = start(rmCallbackHandler)
+      Try({
+        client = start(rmCallbackHandler)
+        sender ! AMRMClientAsyncStartup(Try(true))
```

Comment:
I think this part is also a bit confusing - an activity or sequence diagram would help.
ResourceManagerCallbackHandler is sent by RMCallbackHandlerActor and ResourceManagerCallbackHandlerActor. Both Actors seem very similar - maybe one is for testing only?